clients/src/main/java/org/apache/kafka/common/requests/CreatePartitionsRequest.java

```diff
@@ -68,179 +65,65 @@
         return new Schema[]{CREATE_PARTITIONS_REQUEST_V0, CREATE_PARTITIONS_REQUEST_V1};
     }
 
-    // It is an error for duplicate topics to be present in the request,
-    // so track duplicates here to allow KafkaApis to report per-topic errors.
-    private final Set<String> duplicates;
-    private final Map<String, PartitionDetails> newPartitions;
-    private final int timeout;
-    private final boolean validateOnly;
-
-    public static class PartitionDetails {
-
-        private final int totalCount;
-
-        private final List<List<Integer>> newAssignments;
-
-        public PartitionDetails(int totalCount) {
-            this(totalCount, null);
-        }
-
-        public PartitionDetails(int totalCount, List<List<Integer>> newAssignments) {
-            this.totalCount = totalCount;
-            this.newAssignments = newAssignments;
-        }
-
-        public int totalCount() {
-            return totalCount;
-        }
-
-        public List<List<Integer>> newAssignments() {
-            return newAssignments;
-        }
-
-        @Override
-        public String toString() {
-            return "(totalCount=" + totalCount() + ", newAssignments=" + newAssignments() + ")";
-        }
-
-    }
+    private final CreatePartitionsRequestData data;
+    private final short version;
```

Comment:
I don't think we need this: `AbstractRequest` already stores the version.

Reply:
Thanks for pointing this out. Fixed.